File: app/src/main/java/me/echeung/moemoekyun/api/service/SongsService.java
package me.echeung.moemoekyun.api.service;

import me.echeung.moemoekyun.api.model.response.SongsResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Header;

public interface SongsService {
    @GET("songs")
    Call<SongsResponse> songs(@Header("authorization") String token);
}

File: app/src/main/java/me/echeung/moemoekyun/client/api/v4/service/SongsService.kt
package me.echeung.moemoekyun.client.api.v4.service

import me.echeung.moemoekyun.client.api.v4.ErrorHandlingAdapter
import me.echeung.moemoekyun.client.api.v4.response.SongsResponse
import retrofit2.http.GET
import retrofit2.http.Header

interface SongsService {
    @GET("songs")
    fun getSongs(@Header("Authorization") token: String, @Header("library") library: String): ErrorHandlingAdapter.WrappedCall<SongsResponse>
}
